common/wuanalysis/anawu.hpp

```diff
@@ -20,13 +20,17 @@
 
 #include "anacommon.hpp"
 
-struct WuAnalyseOptions
+typedef enum { wuOptionFirst=0, minInterestingTime=0, minInterestingCost, skewThreshold } WuOptions ;
+
+interface IAnalyserOptions
 {
-    stat_type minInterestingTime = msecs2StatUnits(10);// ignore anything under 10 millisecond
-    stat_type minCost = seconds2StatUnits(30);          // only interested in costs of > 30s
-    stat_type skewThreshold = statSkewPercent(20);     // minimum interesting skew measurment
+public:
+    virtual void setOptionValue(WuOptions opt, __int64 val) = 0;
```

Comment:
minor: the set and apply functions don't need to be (and probably shouldn't be) in this interface.